### PR TITLE
fix(konnect): handle KongRoute conflict on creation

### PR DESCRIPTION
**Pull request by pmalek**
**What this PR does / why we need it**:

Follow up on #749 for `KongRoute`.

### CODE REVIEW

File: controller/konnect/ops/ops.go [hunk 1]
```diff
@@ -419,3 +421,47 @@ func logEntityNotFoundRecreating[
 			"id", id,
 		)
 }
+
+type entityWithID interface {
+	GetID() *string
+}
+
+func fromSliceToSlice[T any](
```
Comment (randmonkey) [inline controller/konnect/ops/ops.go]: Could we some comments and give it a clearer name? `fromSliceToSlice` is really hard to understand.

File: controller/konnect/ops/ops.go [hunk 2]
```diff
@@ -126,7 +126,7 @@ func Create[
 
 	switch {
 	case errors.As(err, &errConflict) ||
-		// Service API returns 400 for conflicts
+		// ControlPlane resource APIs return 400 for conflicts
 		errSdkBody.Code == 3 && errSdkBody.Message == "data constraint error":
```
Comment (randmonkey) [inline controller/konnect/ops/ops.go]: Could we define some function or constant to eliminate the magic number and string here?

**Review by randmonkey — COMMENTED** [review]
Generally LGTM, some minor comments remaining.

File: controller/konnect/ops/ops.go (hunk 2 above)
Comment (pmalek) [reply controller/konnect/ops/ops.go]: f71487ec2e4f0ce959131a05eb51b3486a77d822 PTAL

File: controller/konnect/ops/ops.go (hunk 1 above)
Comment (pmalek) [reply controller/konnect/ops/ops.go]: I've changed this to `sliceToEntityWithIDSlice` and added a comment.

f71487ec2e4f0ce959131a05eb51b3486a77d822

**Review by randmonkey — APPROVED** [review]

**Review by randmonkey — APPROVED** [review]

**Pull request merged**